cfn-resources/federated-database-instance/cmd/resource/resource.go

```diff
@@ -0,0 +1,492 @@
+		return handleError(response, err)
+	}
+	currentModel.getDataLakeTenant(*dataLakeTenant)
+
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "Update Completed",
+		ResourceModel:   currentModel}, nil
+}
+
+// Delete handles the Delete event from the Cloudformation service.
+func Delete(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup()
+
+	modelValidation := validator.ValidateModel(DeleteRequiredFields, currentModel)
+	if modelValidation != nil {
+		return *modelValidation, nil
+	}
+
+	// Create atlas client
+	if currentModel.Profile == nil || *currentModel.Profile == "" {
+		currentModel.Profile = aws.String(profile.DefaultProfile)
+	}
+
+	atlas, peErr := util.NewAtlasClient(&req, currentModel.Profile)
+	if peErr != nil {
+		return *peErr, nil
+	}
+	deleteFederatedDatabaseAPIRequest := atlas.AtlasV2.DataFederationApi.DeleteFederatedDatabase(context.Background(), *currentModel.ProjectId, *currentModel.TenantName)
+	_, response, err := deleteFederatedDatabaseAPIRequest.Execute()
+
+	defer closeResponse(response)
+	if err != nil {
+		_, _ = logger.Warnf("Execute error: %s", err.Error())
+		return handleError(response, err)
+	}
+
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "Delete Completed",
+		ResourceModel:   nil}, nil
+}
+
+// List handles the List event from the Cloudformation service.
+func List(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup()
+
+	modelValidation := validator.ValidateModel(ListRequiredFields, currentModel)
+	if modelValidation != nil {
+		return *modelValidation, nil
+	}
+
+	// Create atlas client
+	if currentModel.Profile == nil || *currentModel.Profile == "" {
+		currentModel.Profile = aws.String(profile.DefaultProfile)
+	}
+
+	atlas, peErr := util.NewAtlasClient(&req, currentModel.Profile)
+	if peErr != nil {
+		return *peErr, nil
+	}
+	listFederatedDatabaseAPIRequest := atlas.AtlasV2.DataFederationApi.ListFederatedDatabases(context.Background(), *currentModel.ProjectId)
+
+	dataLakeTenants, response, err := listFederatedDatabaseAPIRequest.Execute()
+
+	defer closeResponse(response)
+	if err != nil {
+		_, _ = logger.Warnf("Execute error: %s", err.Error())
+		return handleError(response, err)
+	}
+	tenants := make([]interface{}, 0)
+	for i := range dataLakeTenants {
+		var tenant Model
+		tenant.getDataLakeTenant(dataLakeTenants[i])
+		tenants = append(tenants, tenant)
+	}
+
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "List Completed",
+		ResourceModels:  tenants}, nil
+}
+
+func closeResponse(response *http.Response) {
+	if response != nil {
+		response.Body.Close()
+	}
+}
+
+func handleError(response *http.Response, err error) (handler.ProgressEvent, error) {
+	return progress_events.GetFailedEventByResponse(fmt.Sprintf("Error during execution : %s", err.Error()),
```

Comment:
is there a way to add in wich function (CREATE, DELETE, READ , LIST, UPDATE) was the error thrown?? is easy to follow on the testing stage